### DISCUSSION TITLE
gl.users.list() filter attributes - without_projects_bots

### DISCUSSION THREAD

**Discussion by danielhass** [python-gitlab/python-gitlab#2614] [Q&A] [answered]
Dear python-gitlab community and maintainers,

I'm currently searching for a way to make use of the upstream query attributes listed e.g. [here](https://docs.gitlab.com/ee/api/users.html#for-non-administrator-users) with python-gitlab.

I noticed that the docs list a set of _Object listing filters_, see [here](https://python-gitlab.readthedocs.io/en/stable/api/gitlab.v4.html#gitlab.v4.objects.UserManager). Are these a supported or implemented set of upstream attribute filter options?

Or are all upstream options supported out of the box via e.g. `kwargs`?

My experiments so far didn't work:
- `gl.users.list(without_projects_bots="true")`
- `gl.users.list(without_projects_bots=True)`

**Comment by nejch** [comment, marked as answer]
I think there's a typo in the GitLab docs, can you try `gl.users.list(without_project_bots=True)` instead?

https://gitlab.com/gitlab-org/gitlab/-/blob/master/lib/api/users.rb#L100

Edit: but yes, you should be able to use all GitLab parameters like that :)

**Reply by danielhass** [reply]
@nejch thanks for the super fast reply!

Wow.. you are 100% correct 😉 - with the correct parameter name it works as expected. I will open a MR for the GitLab docs to get this fixed.

Just out of curiosity (if you find time to answer), what are the _Object listing filters_ for that are listed in the docs?

**Reply by nejch** [reply]
Great to hear it worked!

The explicitly defined filters are used in our CLI where we can display them in the help and collect from the args. But as you can see that's hard to maintain and keep up to date;)

**Reply by danielhass** [reply]
I see, thanks for the clarification!

I leave the link to the upstream MR that fixes this docs error here for anyone interested: https://gitlab.com/gitlab-org/gitlab/-/merge_requests/126497
